src/coreclr/tools/ILVerify/Program.cs

```diff
@@ -351,6 +348,33 @@ private void VerifyTypes(PEReader peReader, EcmaModule module, string path, ref
             }
         }
 
+        /// <summary>
+        /// Returns full class name, includes parent class for nested class.
+        /// </summary>
+        private string GetFullClassName(MetadataReader metadataReader, TypeDefinitionHandle typeDefinitionHandle)
+        {
+            var typeDef = metadataReader.GetTypeDefinition(typeDefinitionHandle);
+            var typeName = metadataReader.GetString(typeDef.Name);
+
+            var fullName = new StringBuilder();
+            var namespaceName = metadataReader.GetString(typeDef.Namespace);
+
+            var declaringType = typeDef.GetDeclaringType();
+            if (string.IsNullOrEmpty(namespaceName) && !declaringType.IsNil)
+            {
+                fullName.Append(GetFullClassName(metadataReader, declaringType));
+                fullName.Append("/");
```

Comment:
How did you choose `/` for nested type names? Is there some other tool that uses it?

Reply:
AFAIR `+` is the symbol that the runtime uses for printing such types.